lua/weapons/weapon_laserpointer.lua

```diff
@@ -0,0 +1,326 @@
+        end
+    end)
+    timer.Create("BlindnessUpdater", 0.1, 0, function()
+    	for i,v in next, player:GetAll() do
+    		if IsValid(v) then
+    		local blind = v:GetNWFloat("Blindness", 0)
+				if blind ~= nil then
+					if (blind or 0) > 0 then blind = blind-2 end
+					if blind < 0 then blind = 0 end
+
+					v:SetNWFloat("Blindness",blind)
+				end
+			end
+		end
+	end)
+end
+
+function SWEP:DrawWorldModel()
+	local ply = self.Owner
+
+	if(IsValid(ply))then
+		local bn = "ValveBiped.Bip01_R_Hand"
+		local bon = ply:LookupBone(bn) or 0
+
+		local opos = self:GetPos()
+		local oang = self:GetAngles()
+		local bp,ba = ply:GetBonePosition(bon)
+		if(bp)then opos = bp end
+		if(ba)then oang = ba end
+
+		oang:RotateAroundAxis(oang:Right(), -90)
+		oang:RotateAroundAxis(oang:Up(), 90)
+		opos = opos + oang:Forward()*-2
+		opos = opos + oang:Right()*1.25
+		opos = opos + oang:Up()*1.5
+
+		self.laserorgpos:Set(opos)
+		self.laserorgang:Set(oang)
+		self:SetNWVector("laserorgpos",opos)
+		self:SetNWAngle("laserorgang",oang)
+
+		self:SetupBones()
+
+		if IsValid(self) and self:GetClass() == "weapon_laserpointer" and self.LaserBeamOn and self.Weapon:Clip1() > 0  then
+			local tr = self.Owner:GetEyeTrace()
+			local lsrclr = self.LaserColors[self.LaserPower+1]
+
+			cam.Start3D(EyePos(), EyeAngles())
+				local size = math.random()*10
+
+				--render.SetMaterial(Material("sprites/bluelaser1"))
+				render.SetMaterial(Material("cable/new_cable_lit"))
+				render.DrawBeam(self.laserorgpos+self.laserorgang:Up()*6, tr.HitPos, 0.1, 0, 12.5, lsrclr, false)
+				render.SetMaterial(Material("cable/new_cable_lit_back"))
+				lsrclr.a = 0
+				render.DrawBeam(self.laserorgpos+self.laserorgang:Up()*6, tr.HitPos, 0.2, 0, 12.5, lsrclr, false)
+				render.SetMaterial(Material("sprites/light_glow02_add"))
+				lsrclr.a = 255
+				render.DrawQuadEasy(tr.HitPos, tr.HitNormal, size, size, lsrclr)
+			cam.End3D()
+		end
+
+		local mrt = self:GetBoneMatrix(0)
+
+		if(mrt)then
+			mrt:SetTranslation(opos)
+			mrt:SetAngles(oang)
+			self:SetBoneMatrix(0, mrt)
+		end
+	end
+	self:DrawModel()
+end
+
+function SWEP:GetViewModelPosition(pos, ang)
+	pos = pos + ang:Right()*2
+	pos = pos + ang:Up()*-3
+	pos = pos + ang:Forward()*5
+	ang:RotateAroundAxis(ang:Forward(), -3)
+	ang:RotateAroundAxis(ang:Right(), -90)
+	ang:RotateAroundAxis(ang:Up(), -105)
+	pos = pos + ang:Forward()*0.1
+	return pos, ang
+end
+
+function SWEP:PreDrawViewModel()
+	if IsValid(self) and self:GetClass() == "weapon_laserpointer" and self.LaserBeamOn and self.Weapon:Clip1() > 0  then
+		local tr = self.Owner:GetEyeTrace()
+		local vm = self.Owner:GetViewModel()
+		local attachmentIndex = vm:LookupAttachment("tip")
+		local origin = vm:GetAttachment(attachmentIndex).Pos
+		local lsrclr = self.LaserColors[self.LaserPower+1]
+
+		cam.Start3D(EyePos(), EyeAngles())
+			local size = math.random()*10
+
+			render.SetMaterial(Material("sprites/bluelaser1"))

```

Comment:
Factor out  a DrawLaser function with origin and direction as arguments

Reply:
I don't know about this one, I wanted the laser to act as similar to the physgun beam and right now it does. If I I make that change then it'll change it from the original idea I had, but if you want it that way I will change it.

Reply:
why would it act different? I mean just take this code that is copied in 3 different locations and put it in one function, then call that function

Reply:
Will do